### PR TITLE
feat: implement upsert function

### PR DESCRIPTION
**Pull request by OleksiiKachan**
- [x] tests and linter show no problems (`npm t`)
- [x] tests are added/updated for bug fixes and new features
- [x] code is properly formatted (`npm run fmt`)
- [x] description of changes is added in CHANGELOG.md
- [x] update .d.ts typings

added Database.upsert method
added Database.fields method
added Database.tables method
added types

https://github.com/metarhia/metasql/issues/115

### CODE REVIEW

**Review by tshemsedinov — APPROVED** [review]

**Comment by tshemsedinov** [comment]
Do you need help to add typings?

**Comment by OleksiiKachan** [comment]
@tshemsedinov I'm pushing types in different branch
noticed that metasql.d.ts has empty class, so I branched and added declarations

**Comment by OleksiiKachan** [comment]
@tshemsedinov https://github.com/metarhia/metasql/pull/158

**Comment by OleksiiKachan** [comment]
@tshemsedinov added types

File: lib/database.js [hunk 1]
```diff
@@ -50,6 +50,25 @@ const updates = (delta, firstArgIndex = 1) => {
...
+const buildSystemQuery = (type, fields, ...conditions) => {
+  const names = fields[0] === '*' ? '*' : `"${fields.join('", "')}"`;
+  const clause = [];
+  for (const where of conditions) {
+    const keys = Object.keys(where);
+    for (const key of keys) {
+      const [operator, value] = whereValue(where[key]);
+      clause.push(`${key} ${operator} ${value}`);
+    }
+  }
```
Comment (tshemsedinov) [inline lib/database.js:62]: We can use `buildWhere` here

**Review by tshemsedinov — CHANGES_REQUESTED** [review]

**Comment by OleksiiKachan** [comment]
waiting for https://github.com/metarhia/metasql/pull/167

**Pull request closed without merging**